cmd/cosign/cli/copy/copy.go

```diff
@@ -71,6 +72,43 @@ func CopyCmd(ctx context.Context, regOpts options.RegistryOptions, srcImg, dstIm
 		return err
 	}
 
+	idx, isIndex := root.(oci.SignedImageIndex)
+
+	if platform != "" && !isIndex {
```

Comment:
IMO this is clearer as:

```
func RootForPlatform(root, platform string) string, error {
    if platform == "" {
        // Copy all platforms
        return root, nil
    }
    if !isIndex {
        return fmt.Errorf("...")
    }
   // ...
}

Reply:
That code won't compile as-is, there's a couple other things to thread through but I hope it gets the idea across.